controllers/etcdpeer_controller_test.go

```diff
@@ -86,6 +100,12 @@ func (s *controllerSuite) testPeerController(t *testing.T) {
 			etcdPeer.Spec.ClusterName,
 			"Cluster name not set as Pod subdomain")
 
+		expectations := map[string]interface{}{
+			`.spec.template.spec.volumes[?(@.name=="etcd-data")].persistentVolumeClaim.claimName`:              etcdPeer.Name,
+			`.spec.template.spec.containers[?(@.name=="etcd")].volumeMounts[?(@.name=="etcd-data")].mountPath`: "/var/lib/etcd",
+		}
+		try.CheckStructFields(t, expectations, replicaSet)
```

Comment:
I got tired of creating writing for loops and multiple assertions for nested field values, so I wrote `CheckStructFields`. 
I quite like it, especially that you can pick out list elements and map elements easily.
But I'm happy to rewrite in the style of the other tests if you think this is too flaky.

Reply:
i like it, apart from that it pushes type error catching from compile time -> test time. i can't however think of any way around that without some gnarly reflection. 

regardless i'm not sure `try` is the correct package to put it. the package is admittedly lacking a `doc.go` but it was intended to be used as a place to put helpers that needed to try to achieve something multiple times.